src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -214,6 +233,36 @@ public static void PreloadAsmFromPath(string preloaderLocation, string asmLocati
             Debug.WriteLine("Successfully loaded ASM binaries");
         }
 
+        /// <summary>
+        /// Attempts to remap a an old LibG path to a new one using a version map.
+        /// We assume that the leaf directory is of the form LibG_[Version].
+        /// </summary>
+        /// <param name="preloaderLocation"></param>
+        /// <returns> new version LibG path or Empty string if the path could not be remapped.</returns>
+        internal static string RemapOldLibGPathToNewVersionPath(string preloaderLocation)
+        {
+            if (String.IsNullOrEmpty(preloaderLocation))
+            {
+                return string.Empty;
+            }
+            var folderName = Path.GetFileName(preloaderLocation);
+            var splitName = folderName.Split('_');
+            if (splitName.Count() == 2)
```

Comment:
@mjkkirschner so this assumes new versions of LibG will always have 2+ underscores?

ex:
oldVersion = LibG_223
newVersion = LibG_223_0_1

Reply:
yep